cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/server.go

```diff
@@ -172,6 +186,61 @@ func (s *Server) GetAvailablePackages(ctx context.Context, request *corev1.GetAv
 	}, nil
 }
 
+// GetPackageMeta streams the package metadata based on the request.
+func (s *Server) GetPackageMeta(ctx context.Context, request *corev1.GetPackageMetaRequest) (*corev1.GetPackageMetaResponse, error) {
+	log.Infof("+GetPackageMeta(%v)", ctx)
+
+	client, err := s.GetClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	chartsResource := schema.GroupVersionResource{
+		Group:    "source.toolkit.fluxcd.io", //fluxGroup,
+		Version:  "v1beta1",                  //fluxVersion,
+		Resource: "helmcharts"}               //fluxHelmCharts}
+
+	unstructuredChart := unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"apiVersion": "source.toolkit.fluxcd.io/v1beta1", //fmt.Sprintf("%s/%s", fluxGroup, fluxVersion),
+			"kind":       "HelmChart",                        //fluxHelmChart,
+			"metadata": map[string]interface{}{
+				"generateName": "redis-",
+			},
+			"spec": map[string]interface{}{
+				"chart":   "redis",
+				"version": "10.5.x",
+				"sourceRef": map[string]interface{}{
+					"name": "bitnami",
+					"kind": "HelmRepository",
+				},
+				"interval": "10m",
+			},
+		},
+	}
+
+	/*
+		log.Infof("helm charts:")
+		chartList, err := client.Resource(chartsResource).List(ctx, metav1.ListOptions{})
+		if err != nil {
+			return nil, err
+		}
+		for _, chartUnstructured := range chartList.Items {
+			log.Infof("%v", chartUnstructured.Object)
+		}
+	*/
+
+	newChart, err := client.Resource(chartsResource).Create(ctx, &unstructuredChart, metav1.CreateOptions{})
+	if err != nil {
+		log.Errorf("error creating chart: %v\n%v", err, unstructuredChart)
+		return nil, err
+	}
+
+	log.Infof("created chart: [%v]", newChart)
+
+	return nil, status.Errorf(codes.Unimplemented, "not implemented yet")
```

Comment:
Guessing you plan at this point to wait for the chart to reconcile before returning the metadata, though I don't see it on the CR (https://fluxcd.io/docs/components/source/helmcharts/#specification) so we'd need to parse the chart itself, right? (which is cached by fluxv2?)

Reply:
please hold comments until I mark it ready for review :-)